app/src/main/java/ch/epfl/sweng/rps/persistence/Cache.kt

```diff
@@ -1,25 +1,29 @@
-package ch.epfl.sweng.rps.persistance
+package ch.epfl.sweng.rps.persistence
 
 import android.content.Context
+import android.util.Log
+import ch.epfl.sweng.rps.db.Env
 import ch.epfl.sweng.rps.db.FirebaseHelper
 import ch.epfl.sweng.rps.db.FirebaseReferences
 import ch.epfl.sweng.rps.db.FirebaseRepository
 import ch.epfl.sweng.rps.models.User
 import ch.epfl.sweng.rps.models.UserStat
+import ch.epfl.sweng.rps.services.ServiceLocator
 import java.net.InetAddress
 
 
-class Cache private constructor(val ctx:Context, val preferFresh:Boolean = false) {
+class Cache private constructor(private val ctx:Context, val preferFresh:Boolean = false) {
     companion object {
         var cache:Cache? = null
-        fun getInstance(ctx:Context, preferFresh: Boolean = false): Cache {
-            if(cache != null) return cache!!
-            cache = Cache(ctx, preferFresh)
+        fun getInstance(preferFresh: Boolean = false): Cache? {
+            return cache
+        }
+        fun createInstance(ctx : Context ): Cache {
+            cache = Cache(ctx.applicationContext, true)
             return cache!!
         }
     }
-    private val fbRef = FirebaseReferences()
-    private val fbRepo = FirebaseRepository.createInstance(fbRef)
+    private val fbRepo = ServiceLocator.getInstance(Env.Test).repository
```

Comment:
why specify the end ? you shouldn't specify the env usually, as it's usuing the default set one.

Reply:
Oops you're right, it was for testing thought I removed it!